React/NextJs/ProjectStructure.md

```diff
@@ -0,0 +1,212 @@
+
+
+### Layouts
+
+With Layouts, we generally define overall page structure that will be presented to the user based on a specific auth state or a specific page.
+
+For example, your app could have an Admin dashboard on `/admin` route which has a fixed header and sidebars on both right and left, and scrollable content in the middle (you're most likely familiar with this kind of layout).
+
+You will define your different layouts inside `layouts` folder:
+
+```
+.
+.
+└── components
+    └── layouts
+        ├── AdminLayout
+        │   └── index.tsx
+        ├── MainLayout
+        │   └── index.tsx
+        └── BlogLayout
+            └── index.tsx
+```
+
+Then, when creating your routes (pages), you will wrap your page in the layout that represents the current page:
+
+```tsx
+// pages/index.tsx
+
+export default function Home() {
+  return (
+    <MainLayout>
+      ...
+    </MainLayout>
+  )
+}
+
+// pages/admin.tsx
+export default function Admin() {
+  return (
+    <AdminLayout>
+      ...
+    </AdminLayout>
+  )
+}
+```
+
+## Setting up store
+
+Your datx store will be placed in the root of the `src` folder as follows:
+
+```
+src
+└── store
+    ├── models
+    │   ├── User.ts
+    │   └── Session.ts
+    ├── utlis
+    │   ├── config.ts
+    │   └── initializeStore.ts
+    └── index.ts
+```
+
+## Fetchers
+
+In `src/fetchers` you will organize your (swr) fetchers, usually by the model on which you will make API calls.
+
+```ts
+// src/fetchers/user.ts
+
+import { AppCollection } from '../store';
+import { User } from '../store/models/User';
+
+export async function fetchUser(store: AppCollection, id: string): Promise<User> {
+  try {
+    const response = await store.fetch(User, id, {
+      include: ['albums']
+    });
+
+    return response.data as User;
+  } catch(response) {
+    // handle response
+  }
+}
+
+```
+
+You will use your fetcher functions with `useSwr` hook inside of your components (organisms, molecules, pages).
+
+
+## Setting up theming and styles
+
+_Sync with @mamihi_official_
+* Styles will probably be a part of components organization
+
+## Tests
+
+(For PR Review):
+I personally did not do testing that much, so I found inspiration for writing in some projects and blogs.
+I believe this section will be refactored.
+
+### Components
+
+When testing atoms or molecules, you ussualy want to place test file within the same folder:
+```
+└── atoms
+    └── Button
+        ├── __snapshots__
+        ├── index.tsx
+        ├── stories.tsx
+        └── Button.test.tsx
+```
+
+### Utils and hooks
+
+Testing files for hooks and files are gona be placed in a special subfolder so it could be catched when running tests.
+
+```
+  .
+  ├── hooks
+  │   ├── __tests__
+  │   │   ├── useModal.test.ts
+  │   │   └── useNotification.test.ts
+  │   ├── useModal.ts
+  │   └── useNotification.ts
+  └── utils
+      ├── __tests__
+      │   └── generateFileTree.test.ts
+      └── generateFileTree.ts
+
+```
```

Comment:
Maybe a recap would be useful here, as a more complex example that includes most of the covered cases?